src/test/java/io/bf2/kafka/authorizer/CustomAclAuthorizerTest.java

```diff
@@ -203,76 +202,47 @@ void testAuthorize(String title,
-
-            AuthorizableRequestContext rc = Mockito.mock(AuthorizableRequestContext.class);
-            Mockito.when(rc.clientAddress()).thenReturn(InetAddress.getLoopbackAddress());
-            Mockito.when(rc.listenerName()).thenReturn("security-9095");
-            Mockito.when(rc.principal()).thenReturn(new KafkaPrincipal(KafkaPrincipal.USER_TYPE, "owner1"));
-            Mockito.when(rc.requestType()).thenReturn((int) ApiKeys.CREATE_ACLS.id);
-
-            AclBinding readUser1Topics = new AclBinding(
-                    new ResourcePattern(ResourceType.CLUSTER, "my-cluster", PatternType.LITERAL),
-                    new AccessControlEntry("User:user1", "*", AclOperation.CLUSTER_ACTION, AclPermissionType.ALLOW));
-            AclBinding writeUser1Topics = new AclBinding(
-                    new ResourcePattern(ResourceType.CLUSTER, "my-cluster", PatternType.LITERAL),
-                    new AccessControlEntry("User:user1", "*", AclOperation.IDEMPOTENT_WRITE, AclPermissionType.ALLOW));
-            AclBinding describleAllDelegationTokens = new AclBinding(
-                    new ResourcePattern(ResourceType.DELEGATION_TOKEN, "*", PatternType.LITERAL),
-                    new AccessControlEntry("User:user1", "*", AclOperation.DESCRIBE, AclPermissionType.ALLOW));
-
-            var bindings = List.of(readUser1Topics, writeUser1Topics, describleAllDelegationTokens);
-            var results = auth.createAcls(rc, bindings);
-
-            assertEquals(3, results.size());
-            assertTrue(results.stream()
-                    .map(CompletionStage::toCompletableFuture)
-                    .map(CompletableFuture::join)
-                    .map(AclCreateResult::exception)
-                    .map(Optional::get)
-                    .allMatch(e -> e instanceof ApiException
-                            && CustomAclAuthorizer.CREATE_ACL_INVALID_BINDING.equals(e.getMessage())));
-        }
+        AclBinding readUser1Topics = new AclBinding(
+                new ResourcePattern(ResourceType.CLUSTER, "my-cluster", PatternType.LITERAL),
+                new AccessControlEntry("User:user1", "*", AclOperation.CLUSTER_ACTION, AclPermissionType.ALLOW));
+        AclBinding writeUser1Topics = new AclBinding(
+                new ResourcePattern(ResourceType.CLUSTER, "my-cluster", PatternType.LITERAL),
+                new AccessControlEntry("User:user1", "*", AclOperation.IDEMPOTENT_WRITE, AclPermissionType.ALLOW));
+        AclBinding describleAllDelegationTokens = new AclBinding(
+                new ResourcePattern(ResourceType.DELEGATION_TOKEN, "*", PatternType.LITERAL),
+                new AccessControlEntry("User:user1", "*", AclOperation.DESCRIBE, AclPermissionType.ALLOW));
+        var bindings = List.of(readUser1Topics, writeUser1Topics, describleAllDelegationTokens);
+        createAclsAndExpect("owner1", bindings, (aclCreateResult) -> isFailedWithError(aclCreateResult, CustomAclAuthorizer.CREATE_ACL_INVALID_BINDING), 3);
```

Comment:
Nice refactor! But we validated the error is instanceof ApiException, now, we skipped that. Is that a miss or intended? And why?

Reply:
The signature is `Optional<ApiException> exception()` so the type check was redundant.

Reply:
Cool! Make sense.